scripts/all_the_kinds.py

```diff
@@ -0,0 +1,288 @@
+from collections import namedtuple
+
+
+# we don't have data to evaluate the function, so this just looks up the
+# minimum possible value the function could return.
+KNOWN_FUNCS = {
+    'mz_calculate_path_major_route': 8,
+    'mz_get_min_zoom_highway_level_gate': 14,
+    'tz_looks_like_service_area': 13,
+    'tz_looks_like_rest_area': 13,
+    'mz_calculate_ferry_level': 8,
+}
+
+
+# structures to return data from the parsing functions. basically a key->value
+# mapping where the keys are (layer, kind) tuples and the values are the other
+# stuff we parsed out of the various YAML/CSV files.
+KindKey = namedtuple('KindKey', 'layer kind')
+KindInfo = namedtuple('KindInfo', 'min_zoom sort_rank')
+
+
+# parses the values for a column out of a filter definition. this is used when
+# the output defines "kind: {col: some_column}", and we want to find out what
+# values "some_column" could have.
+#
+# this is most often seen in things such as:
+#
+#  - filter:
+#      amenity: [bus_station, car_rental, recycling, shelter]
+#    min_zoom: 16
+#    output:
+#      <<: *output_properties
+#      kind: {col: amenity}
+#
+# where we've white-listed the amenity values in the filter, so we know all the
+# possible values that the kind can have.
+#
+# theoretically, this can be arbitrarily complex, but it turns out that in all
+# the cases we have at the time of writing, the whitelist is available at the
+# top level of the filter! lucky us :-)
+def parse_filter_for_col(col, filter_defn):
+    if col.startswith('tags->'):
+        raise AssertionError("Output kind column starts with deprecated "
+                             "'tags->' prefix: %r" % (col,))
```

Comment:
I see we've decided to bite the bullet here! 🎉

Reply:
The test suite seems large enough now that I thought we'd probably catch a regression. There's a lot of noise from the change, but at least it's not part of a "feature PR".